CHANGELOG.md

```diff
@@ -5,6 +5,53 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 (modification: no type change headlines) and this project adheres to 
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [2.4.0] - 2018-07-27
+
+With the ``2.4.x`` release series we now start to gradually add ``Constantinople`` features with the
+bitwise shifting instructions from [EIP 145](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-145.md)
+making the start being introduced in the ``v2.4.0`` release.
+
+Since both the scope of the ``Constantinople`` hardfork as well as the state of at least some of the EIPs
+to be included are not yet finalized, this is only meant for ``EXPERIMENTAL`` purposes, e.g. for developer
+tools to give users early access and make themself familiar with dedicated features.
+
+Once scope and EIPs from ``Constantinople`` are final we will target a ``v2.5.0`` release which will officially
+introduce ``Constantinople`` support with all the changes bundled together.
+
+Note that from this release on we also introduce new ``chain`` (default: ``mainnet``) and ``hardfork`` 
+(default: ``byzantium``) initialization parameters, which make use of our new [ethereumjs-common](https://github.com/ethereumjs/ethereumjs-common) library and in the future will allow
+for parallel hardfork support from ``Byzantium`` onwards.
+
+Since ``hardfork`` default might be changed or dropped in future releases, you might want to explicitly
+set this to ``byzantium`` on your next update to avoid future unexpected behavior.
+
+All the changes from this release:
+
+**FEATURES/FUNCTIONALITY**
+
+- Improved chain and fork support, see PR [#304](https://github.com/ethereumjs/ethereumjs-vm/pull/304)
+- Support for the ``Constantinople`` bitwise shifiting instructions ``SHL``, ``SHR`` and ``SAR``, see PR [#251](https://github.com/ethereumjs/ethereumjs-vm/pull/251)
+- New ``newContract`` event which can be used to do interrupting tasks on contract/address creation, see PR [#306](https://github.com/ethereumjs/ethereumjs-vm/pull/306)
+- Alignment of behavior of bloom filter hashing to go along with mainnet compatible clients *BREAKING*, see PR [#295](https://github.com/ethereumjs/ethereumjs-vm/pull/295) 
+
+**UPDATES/TESTING**
+
+- Usage of the latest ``rustbn.js`` API, see PR [#312](https://github.com/ethereumjs/ethereumjs-vm/pull/312)
+- Some cleanup in precompile error handling, see PR [#318](https://github.com/ethereumjs/ethereumjs-vm/pull/318)
+- Some cleanup for ``StateManager``, see PR [#266](https://github.com/ethereumjs/ethereumjs-vm/pull/266)
+- Renaming of ``util.sha3`` usages to ``util.keccak256`` and bump ``ethereumjs-util`` to ``v5.2.0`` (you should do to if you use ``ethereumjs-util``)
```

Comment:
to -> too

Reply:
Ah, great, just wanted to ask: if you now could have a look and would give it a "go", I could do a release and a reddit post on top, which would suite my work flow for today.

I would then tomorrow prepare different ``Constantinople`` issue descriptions and do a "Call for participation" on Reddit for Constantinople dev tomorrow.

Reply:
Hi @axic, please forgive me to not hold up to my own deadline but I would do a release in the next 15 min, otherwise this doesn't play out with the ``Constantinople`` announcement tomorrow and I am a bit afraid that I am the "out-of-office" and this is delayed by weeks.

If you have larger concerns hold me back.